aspnetcore/security/ip-safelist.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Client IP safelist for ASP.NET Core
+author: damienbod
+description: Learn how to write Middleware or action filters to validate remote IP addresses against a list of approved IP addresses.
+ms.author: tdykstra
+ms.custom: mvc
+ms.date: 08/31/2018
+uid: security/ip-safelist
+---
+# Client IP safelist for ASP.NET Core
+
+By [Damien Bowden](https://twitter.com/damien_bod) and [Tom Dykstra](https://github.com/tdykstra)
+ 
+This article shows two ways to implement an IP safelist (also known as a whitelist):
+
+* By using ASP.NET Core middleware to check the remote IP address of every request.
+* By using ASP.NET Core action filters to check the remote IP address of requests for specific action methods.
+
+The sample app illustrates both approaches. In each case, a string containing approved client IP addresses is stored in an app setting. The middleware or filter parses the string into a list and  checks if the remote IP is in the list. If not, an HTTP 403 Forbidden status code is returned.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/security/ip-safelist/samples/2.x/ClientIpAspNetCore) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## The safelist
+
+The list is configured in the *appsettings.json* file. It's a semicolon-delimited list and can contain IPv4 and IPv6 addresses.
```

Comment:
Consider/ Maybe remove from app setting deatail above and write this as
The list is contained in the *app.json* file, a real app would use a database or other data store. 
Or leave as is.